animation_nodes/nodes/midi/noteInfo.py

```diff
@@ -0,0 +1,45 @@
+import bpy
+from ... base_types import AnimationNode, ListTypeSelectorSocket, VectorizedSocket
+from ... data_structures import DoubleList
+
+class midiNoteNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_MidiNoteInfoNode"
+    bl_label = "MIDI Note Info"
+    bl_width_default = 180
+
+    useNotesList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput(VectorizedSocket("midiNote", "useNotesList",
+            ("note", "note"), ("notes", "notes")))
+        self.newOutput(VectorizedSocket("Integer", "useNotesList",
+            ("Channel", "channel"), ("Channels", "channels")))
+        self.newOutput(VectorizedSocket("Integer", "useNotesList",
+            ("note", "note"), ("notes", "notes")))
+        self.newOutput(VectorizedSocket("Float", "useNotesList",
+            ("Time On", "timeon"), ("Times On", "timeson")))
+        self.newOutput(VectorizedSocket("Float", "useNotesList",
+            ("Time Off", "timeoff"), ("Times Off", "timesoff")))
+        self.newOutput(VectorizedSocket("Float", "useNotesList",
+            ("Velocity", "velocity"), ("Velocities", "velocities")))
+
+    def execute(self, note):
+        if self.useNotesList:
+            if note is None:
```

Comment:
Note here would be a list, so you should check if it is an empty lists instead. However, if it is empty, then the `DoubleList.fromValues` method will return an empty list as well, so this check can be skipped.

Reply:
I wait for your choice on this

Reply:
I meant that this whole block is not needed, it should be removed. Those lines between `if note is None:` and `else:` inclusive.